spec/presenters/two_factor_authentication/selection_presenter_spec.rb

```diff
@@ -22,33 +24,105 @@ def method
   end
 
   describe '#disabled?' do
-    it { expect(presenter.disabled?).to eq(false) }
+    let(:single_configuration_only) {}
+    let(:mfa_configuration_count) {}
+
+    before do
+      allow(presenter).to receive(:single_configuration_only?).and_return(single_configuration_only)
+      allow(presenter).to receive(:mfa_configuration_count).and_return(mfa_configuration_count)
+    end
+
+    context 'without single configuration restriction' do
+      let(:single_configuration_only) { false }
+
+      it 'is an mfa that allows multiple configurations' do
+        expect(presenter.disabled?).to eq(false)
+      end
+    end
+
+    context 'with single configuration only' do
+      let(:single_configuration_only) { true }
+
+      context 'with default mfa count implementation' do
+        before do
+          allow(presenter).to receive(:mfa_configuration_count).and_call_original
+        end
+
+        it 'is mfa with unimplemented mfa count and single config' do
+          expect(presenter.disabled?).to eq(false)
+        end
+      end
+
+      context 'with no configured mfas' do
+        let(:mfa_configuration_count) { 0 }
+
+        it 'is configured with no mfa' do
+          expect(presenter.disabled?).to eq(false)
+        end
+      end
+
+      context 'with at least one configured mfa' do
+        let(:mfa_configuration_count) { 1 }
+
+        it 'is mfa with at least one configured' do
+          expect(presenter.disabled?).to eq(true)
+        end
+      end
+    end
```

Comment:
I'd suggest adding test coverage for the new logic branch we added in 981b926

```suggestion
    end

    context 'with configuration' do
      let(:single_configuration_only) { true }
      let(:mfa_configuration_count) { 1 }
      let(:configuration) { create(:phone_configuration, user: user) }

      before do
        allow(presenter).to receive(:configuration).and_return(configuration)
      end

      it { expect(presenter.disabled?).to eq(false) }
    end
```